### PR TITLE
FIX Issue 1005 API v2: Strings are not properly rendered

### PR DESCRIPTION
**Pull request by jmcanterafonseca**
Work in progress patch

### CODE REVIEW

File: src/lib/ngsi/ContextAttribute.cpp [hunk 1]
```diff
@@ -357,18 +357,28 @@ std::string ContextAttribute::toJson(bool isLastElement)
...
   bool         isNumber = false;
+  bool         isString = true;
 
   if (type == "number")
   {
     isNumber = true;
   }
 
-  if (((type == "") || (isNumber == true)) && (metadataVector.size() == 0))
+  if(type == "string")
```
Comment (fgalan) [inline src/lib/ngsi/ContextAttribute.cpp]: Do we really need special treatment for "string" type? As far as I undersrtand, each type that is not a "number" (or a "date") is by default considered string type from a rendering point of view.

File: src/lib/ngsi/ContextAttribute.cpp [hunk 2]
```diff
@@ -357,18 +357,28 @@ std::string ContextAttribute::toJson(bool isLastElement)
 {
   std::string  out;
   bool         isNumber = false;
+  bool         isString = true;
 
   if (type == "number")
```
Comment (fgalan) [inline src/lib/ngsi/ContextAttribute.cpp]: Thus, the type for number is exactly "number" (I mean, "Number" is not a valid numeric type). It is fine... just to check that in our documentation we make clear that "number" is case-sensitive.

File: test/functionalTest/cases/1005_GET_v2_entities_string/1005_GET_v2_entities_string.test [hunk 3]
```diff
@@ -0,0 +1,137 @@
+# Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/functionalTest/cases/1005_GET_v2_entities_string/1005_GET_v2_entities_string.test]: I should  have though in four digits numbers when we start numbering functional test cases.... Probably an overall renaming (e,g. 766_XXX to 0766_X) now would make sense (although not as part of this PR, of course).

Or maybe I'm just delaying the problem and in a couple of month we would reach issue number #10000 :D

NTC

File: test/functionalTest/cases/1005_GET_v2_entities_string/1005_GET_v2_entities_string.test [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      "type": "Car",
+      "attributes" : [
+        {
+          "name" : "speed",
+          "type": "number",
+          "value" : 150
+        },
+        {
+          "name" : "brand",
+          "type": "string",
```
Comment (fgalan) [inline test/functionalTest/cases/1005_GET_v2_entities_string/1005_GET_v2_entities_string.test]: Maybe some of the attributes should use an "implicit string" type, in order to check that it is also correctly rendered. E.g: `"type": "myString"`.

File: test/functionalTest/cases/1005_GET_v2_entities_string/1005_GET_v2_entities_string.test (hunk 3 above)
Comment (fgalan) [inline test/functionalTest/cases/1005_GET_v2_entities_string/1005_GET_v2_entities_string.test]: Actually deferred to tech debt issue: https://github.com/telefonicaid/fiware-orion/issues/1010

Comment (jmcanterafonseca) [inline test/functionalTest/cases/1005_GET_v2_entities_string/1005_GET_v2_entities_string.test]: cool

File: test/functionalTest/cases/1005_GET_v2_entities_string/1005_GET_v2_entities_string.test (hunk 4 above)
Comment (jmcanterafonseca) [inline test/functionalTest/cases/1005_GET_v2_entities_string/1005_GET_v2_entities_string.test]: Fixed in 002cd1a2337d77ca1131cfbdd09f70952486039a

File: src/lib/ngsi/ContextAttribute.cpp (hunk 1 above)
Comment (jmcanterafonseca) [inline src/lib/ngsi/ContextAttribute.cpp]: We are going to have booleans as well. 

I think the current rendering code could be simplified a lot , even we could consider using switch case avoiding too many ifs but that could be part of another PR IMHO and when Ken is here 

NTC

File: src/lib/ngsi/ContextAttribute.cpp (hunk 2 above)
Comment (jmcanterafonseca) [inline src/lib/ngsi/ContextAttribute.cpp]: yup, I would prefer to be consistent with JSON types

File: src/lib/ngsi/ContextAttribute.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/ngsi/ContextAttribute.cpp]: If you think that part is fixable, I'd recommend to hadd a comment over these lines:

```
//FIXME P10: rendering code could be simplified a lot. To discuss with Ken, once he is back
```

or something like that. Otherwise, we are in risk of forgetting thinks (it had happended in the past ;)

File: src/lib/ngsi/ContextAttribute.cpp (hunk 2 above)
Comment (fgalan) [inline src/lib/ngsi/ContextAttribute.cpp]: NTC

**Comment by fgalan** [comment]
LGTM
